extension/bearertokenauthextension/bearertokenauth.go

```diff
@@ -171,6 +174,38 @@ func (b *BearerTokenAuth) RoundTripper(base http.RoundTripper) (http.RoundTrippe
 	}, nil
 }
 
+// Authenticate checks whether the given context contains valid auth data.
+func (b *BearerTokenAuth) Authenticate(ctx context.Context, headers map[string][]string) (context.Context, error) {
+	metadata := client.NewMetadata(headers)
+	authHeaders := metadata.Get("authorization")
+	if len(authHeaders) == 0 {
+		return ctx, errors.New("authentication didn't succeed")
+	}
+
+	// we only use the first header, if multiple values exist
+	parts := strings.Split(authHeaders[0], " ")
+
+	errInvalidAuthorization := errors.New("invalid authorization header format")
```

Comment:
This will allocate a new error value on each invocation, this should be moved to the type of the file as a global var (since it can't be constant since it is not a primitive type).